engine/runtime/src/main/java/org/enso/interpreter/runtime/ThreadExecutors.java

```diff
@@ -20,13 +18,28 @@ final class ThreadExecutors {
     this.context = context;
   }
 
-  final ExecutorService newCachedThreadPool(String name, boolean systemThread) {
+  ExecutorService newCachedThreadPool(String name, boolean systemThread) {
     var s = Executors.newCachedThreadPool(new Factory(name, systemThread));
     pools.put(s, name);
     return s;
   }
 
-  final ExecutorService newFixedThreadPool(int cnt, String name, boolean systemThread) {
```

Comment:
Why do we remove `final`? We don't want anyone to ever override these methods.

Reply:
Class is already final. IDE suggested that

Reply:
I'd suggest to disable this rule in your IDE. There is nothing wrong on making sure _methods aren't virtual_ multiple times.